modules/feeds_migrate_ui/src/Form/MigrationMappingFormBase.php

```diff
@@ -322,11 +332,59 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function copyFormValuesToEntity(EntityInterface $entity, array $form, FormStateInterface $form_state) {
+    // Add the mapping to the process section.
     $mapping = $this->mapping;
     $process = $entity->get('process') ?: [];
-    $process = array_merge($process, $this->migrationEntityHelper()->processMapping($mapping));
+    $process = array_merge($process, $this->migrationEntityHelper()
+      ->processMapping($mapping));
 
     $entity->set('process', $process);
+
+    // Add the unique values to the source section.
+    $source = $entity->get('source');
+    $ids = $source['ids'] ?: [];
+    if ($this->unique) {
+      // Is unique, make sure it's there.
+      if (!array_key_exists($mapping["source"], $ids)) {
+        // Doesn't exist, so add it.
+        $ids[$mapping["source"]] = ['type' => 'string'];
+      }
+    }
+    else {
+      // Is not unique, make sure it's not there.
+      if (array_key_exists($mapping["source"], $ids)) {
+        // Doesn't exist, so add it.
+        unset($ids[$mapping["source"]]);
+      }
+    }
+    $source['ids'] = $ids;
+
+    // Add the fields to the source section.
+    //    fields:
+    //      -
+    //      name: title
+    //      label: Title
+    //      selector: title
+
+    $fields = $source['fields'] ?: [];
+    $foundField = FALSE;
```

Comment:
Nit, we should standardize on `camelCase` versus `snake_case` for local variables at some point. 🤷‍♂️

Reply:
I do `camelCase` in class and `snake_case` in a module file. I think that's the standard, right? :)

Reply:
Sounds good!